### DISCUSSION TITLE
Does Stride Engine plan to add a gameplay system like Unreal Engine?

### DISCUSSION THREAD

**Discussion by CeSun** [stride3d/stride#1552] [Ideas]
The gameplay system is very important for small development teams or individuals. The smaller the team, the more difficult it is to make reliable online functions and reliable game frameworks. Small teams generally prefer to focus on gameplay.

Unreal Engine is very convenient to make an online room game thanks to its gameplay system, but it is too heavyweight. Few new engines on the market will join the gameplay system, only some traditional engines will.

**Comment by manio143** [comment]
I believe this is what you're referring to: https://docs.unrealengine.com/5.0/en-US/gameplay-framework-in-unreal-engine/
It's a high level set of APIs against which one would write gameplay logic. In my opinion this belongs in a plugin rather than the base engine (once we enable creating plugins).

Reminder - Stride is developed by the community, so someone would need to decide to make such a framework on top of Stride and make it available for use to others.

**Reply by CeSun** [reply]
So, anyone can submit high-level apis related to gameplay, right? Unfortunately, my programming skills are too bad.
